### PR TITLE
(CDAP-4661) Allow system namespace in metadata APIs. Also support cas…

### PR DESCRIPTION
**Pull request by bdmogal**
…e-insensitive matching for metadata scope

Jira: [CDAP-4661](https://issues.cask.co/browse/CDAP-4661)
Build: http://builds.cask.co/browse/CDAP-RBT604

### CODE REVIEW

File: cdap-app-fabric/src/main/java/co/cask/cdap/metadata/EntityValidator.java [hunk 1]
```diff
@@ -65,7 +65,9 @@ public void ensureEntityExists(Id.NamespacedId entityId) throws NotFoundExceptio
     try {
       namespaceClient.get(entityId.getNamespace());
     } catch (NamespaceNotFoundException e) {
-      throw e;
+      if (!Id.Namespace.SYSTEM.equals(entityId.getNamespace())) {
```
Comment (anwar6953) [inline cdap-app-fabric/src/main/java/co/cask/cdap/metadata/EntityValidator.java]: why not check for System namespace and altogether skip the nsClient#get call for efficiency.

Comment (bdmogal) [inline cdap-app-fabric/src/main/java/co/cask/cdap/metadata/EntityValidator.java]: fixed

**Comment by poornachandra** [comment]
Changes LGTM. Can we add a test for this?

**Comment by bdmogal** [comment]
@poornachandra have added tests in the last commit. Didn't have them before because there was no way of adding system-scope artifacts via REST APIs, but figured out a way of doing so. Please review again when you have time.

**Comment by poornachandra** [comment]
@bdmogal Thanks for adding the test! Please squash and merge.

**Comment by bdmogal** [comment]
Squashed commits. Build green. Merging.

**Pull request merged**